File: bridget_mcmahon/week11/monday/int_value/int.rb
# Find largest integer in an array

a = Array.new(20) { rand(1..100) }

##### ITERATION #####
def find_largest_integer(arr)
  largest = -1.0 / 0.0 # Negative Infinity

  arr.each do |n|
    largest = n if n > largest
  end

  largest
end

##### REDUCE / INJECT #####


##### RECURSION #####


p a
puts find_largest_integer(a)

File: bridget_mcmahon/week11/monday/int_value/int.js
// Find the largest int value in an array

let a = [1, 7, 6, 12, 3, 30, 42, 19, 22, 9];

// Iteration
const findLargestInteger = (arr) => {
  let largest = 0;

  for (let i = 0; i < arr.length; i++) {
    if (arr[i] > largest) {
      largest = arr[i];
    }
  }
  return largest;
};

const findLargestInteger2 = (arr) => {
  let max = -Infinity;

  arr.forEach((n) => {
    if (n > max) max = n;
  });

  return max;
};


// Reduce
const largest = (arr) => {
  return arr.reduce( (max, n) => n > max ? n : max);
};

// const largest = (arr) => {
//   return arr.reduce( (max, n) => {
//     if (n > max) {
//       return n;
//     } else {
//       return max;
//     }
//   });
// };

console.log(findLargestInteger(a));
console.log(findLargestInteger2(a));
console.log(largest(a));
